app/src/main/java/org/thoughtcrime/securesms/onboarding/LandingActivity.kt

```diff
@@ -19,28 +83,185 @@ class LandingActivity : BaseActionBarActivity() {
         KeyCachingService.setMasterSecret(applicationContext, Object())
     }
 
-    private fun register() {
-        val intent = Intent(this, RegisterActivity::class.java)
-        push(intent)
+    @Preview
+    @Composable
+    private fun LandingScreen(
+        @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+    ) {
+        PreviewTheme(themeResId) {
+            LandingScreen()
+        }
+    }
+
+    @Composable
+    private fun LandingScreen() {
+        var count by remember { mutableStateOf(0) }
+        val listState = rememberLazyListState()
+
+        LaunchedEffect(Unit) {
+            delay(500.milliseconds)
+            while(count < MESSAGES.size) {
+                count += 1
+                listState.animateScrollToItem(0.coerceAtLeast((count - 1)))
+                delay(1500L)
+            }
+        }
+
+        Column(modifier = Modifier.padding(horizontal = 36.dp)) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                stringResource(R.string.onboardingBubblePrivacyInYourPocket),
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                style = MaterialTheme.typography.h4,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(24.dp))
+
+            LazyColumn(
+                state = listState,
+                modifier = Modifier
+                    .heightIn(min = 200.dp)
+                    .fillMaxWidth()
+                    .weight(2f),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+                items(
+                    MESSAGES.take(count),
+                    key = { it.stringId }
+                ) { item ->
+                    AnimateMessageText(
+                        stringResource(item.stringId),
+                        item.isOutgoing
+                    )
+                }
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            OutlineButton(
+                text = stringResource(R.string.onboardingAccountCreate),
+                modifier = Modifier
+                    .contentDescription(R.string.AccessibilityId_create_account_button)
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                onClick = ::startPickDisplayNameActivity
+            )
+            Spacer(modifier = Modifier.height(14.dp))
+            FilledButton(
+                text = stringResource(R.string.onboardingAccountExists),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally)
+                    .contentDescription(R.string.AccessibilityId_restore_account_button)
+            ) { start<LinkDeviceActivity>() }
+            Spacer(modifier = Modifier.height(8.dp))
+            BorderlessButton(
```

Comment:
There's a lot of very specific values hardcoded here ( the width, the font size, the lineheight). Is this button reused anywhere? Shouldn't it be made as a reusable component?
Also all the values regarding typography should be extracted into an organised Type file and accessed appropriately instead of hardcoded each time into every composables.
We might need to do an overall styling pass on all our composables later, because I think it has been done everywhere so far.

I have seen this exact same code in Ryan's PR too, so it's being reused.

Might be hard to rework now, but we need to be mindful of this for future composables.

Reply:
Font sizes and line heights are taken from the style now.